.github/actions/install-pre-commit/action.yml

```diff
@@ -61,6 +61,16 @@ runs:
         key: cache-pre-commit-v4-${{ inputs.python-version }}-${{ hashFiles('.pre-commit-config.yaml') }}
         path: /tmp/
       id: restore-pre-commit-cache
+    - name: "Check if pre-commit cache tarball exists"
```

Comment:
Actually now when I think of it - we could also skip the whole "Restore .cache from the tar file" step if "restore-pre-commit-cache" fails - that would be slightly slower but should still work. @amoghrajesh

Reply:
Makes sense. Ill look

Reply:
@potiuk came back at this one. If the `steps.restore-pre-commit-cache.outputs.stash-hit ` was false, it would anyway skip that step below, but due to some condition the flag was true but the file was not present, making this pre validate essential. (check linked CI failure)

Makes sense or not?